src/main/webapp/app/exercises/programming/shared/lifecycle/programming-exercise-lifecycle.component.ts

```diff
@@ -82,4 +86,20 @@ export class ProgrammingExerciseLifecycleComponent implements OnInit {
             alert(this.translator.instant('artemisApp.programmingExercise.timeline.alertNewAfterDueDate'));
         }
     }
+
+    /**
+     * Updates the example solution publication date of the programming exercise if it is set and not after release or due date.
+     * Due date check is not performed if exercise is not included in the grade.
+     * @param newReleaseOrDueDate the new exampleSolutionPublicationDate if it is after the current exampleSolutionPublicationDate
+     */
+    updateExampleSolutionPublicationDate(newReleaseOrDueDate?: dayjs.Dayjs) {
+        if (this.exerciseService.hasExampleSolutionPublicationDateError(this.exercise)) {
+            const message =
+                this.exercise.dueDate != undefined
+                    ? 'artemisApp.programmingExercise.timeline.alertNewExampleSolutionPublicationDateAsDueDate'
+                    : 'artemisApp.programmingExercise.timeline.alertNewExampleSolutionPublicationDateAsReleaseDate';
+            alert(this.translator.instant(message));
```

Comment:
this should probably be an alertservice info popup instead of a js alert to not just stop everything

Reply:
`updateDueDate` method above that one also used `alert()` so I wanted to make it consistent with it. Also, since the date is changed after the user closes the alert box, this gives them a chance to look at the previous value of the date that will be changed. Being able to do a last manual check can be convenient for the user  in my opinion.

https://github.com/ls1intum/Artemis/blob/cda2dbdf339f98e3f5cdcfb66cd5a66c9b57be43/src/main/webapp/app/exercises/programming/shared/lifecycle/programming-exercise-lifecycle.component.ts#L78-L88